impeller/aiks/canvas.cc

```diff
@@ -725,6 +725,14 @@ void Canvas::SaveLayer(const Paint& paint,
   auto& new_layer_pass = GetCurrentPass();
   new_layer_pass.SetBoundsLimit(bounds);
 
+  if (paint.image_filter) {
+    MipCountVisitor mip_count_visitor;
+    paint.image_filter->Visit(mip_count_visitor);
+    new_layer_pass.SetRequiredMipCount(
+        std::max(mip_count_visitor.GetRequiredMipCount(),
+                 new_layer_pass.GetRequiredMipCount()));
```

Comment:
The new layer will always have a required mip count of 0. The place where the maxing needs to happen is when a layer is being appended that has a backdrop filter.

Reply:
> The place where the maxing needs to happen is when a layer is being appended that has a backdrop filter.

That still exists from the last PR.  This covers both cases, image filter and backdrop filter.  Is there a case this doesn't cover, or are you suggesting there is a singular place we can put this code instead of having it in 2 locations?

Reply:
Wait, you are saying the max needs to happen there as well?  I'm not understanding the order of operations that is problematic.  Can you lay it out for me?

Reply:
The mip count we need to use when allocating the texture is the maximum we detect for all reads. A backdrop texture is read by both subpass backdrop filters as well as the image filter that happened to be set at the time the SaveLayer was made.

Inside of Save(), the RequiredMipCount of the new layer is being set to the backdrop filter's mip count. But backdrop filter will read from the parent layer's backdrop, not the new layer's backdrop.

Reply:
So when we come across a backdrop filter, we need to max the mip count with the _parent_ pass mip count. And when an image filter is being applied via the paint state, no maxing is necessary because nothing should have set the mip count yet.

Reply:
Ah yea I see, thanks.  This started making the previous tests print out the "no mips" message when rendering the blur.  I would like to have a way to assert that in the tests.  We still have some tests that use the filter without mips (assuming they will have them at runtime) so I was a bit squishy with that.  I'll see if I can find a way to assert it.

Reply:
Okay fixed.  I added a logs check to make sure this doesn't regress.  We don't have a better way to assert this currently.  I can file an issue to make it a hard assert and migrate all the tests over.